src/__tests__/parser.test.ts

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable @typescript-eslint/no-unused-expressions */
+import { expect } from "chai";
+import { TokenMap, fetchTokenList } from "src/hooks/tokenList";
+import { TokenInfo } from "@uniswap/token-lists";
+import { parseCSV } from "src/parser";
+import { SafeInfo } from "@gnosis.pm/safe-apps-sdk";
+import BigNumber from "bignumber.js";
+
+let tokenList: TokenMap;
+let listedTokens: string[];
+let listedToken: TokenInfo;
+const unlistedTokenAddress: string =
+  "0x6b175474e89094c44da98b954eedeac495271d0f";
+
+let validReceiverAddress = "0x1000000000000000000000000000000000000000";
+
+const headerRow = "token_address,receiver,amount,decimals";
+
+let dummySafeInfo: SafeInfo = {
+  safeAddress: "0x123",
+  network: "rinkeby",
+  ethBalance: "100",
+};
+
+describe("Parsing CSVs ", () => {
+  beforeEach(async () => {
+    tokenList = await fetchTokenList(dummySafeInfo.network);
+    listedTokens = Array.from(tokenList.keys());
+    listedToken = tokenList.get(listedTokens[0]);
+  });
+  it("should transform simple, valid CSVs correctly", async () => {
+    const rowWithoutDecimal = [listedToken.address, validReceiverAddress, "1"];
+    const rowWithDecimal = [
+      unlistedTokenAddress,
+      validReceiverAddress,
+      "69.420",
+      "18",
+    ];
+    const rowWithoutTokenAddress = ["", validReceiverAddress, "1"];
+
+    const [payment, warnings] = await parseCSV(
+      headerRow +
+        "\n" +
+        rowWithoutDecimal.join(",") +
+        "\n" +
+        rowWithDecimal.join(",") +
+        "\n" +
+        rowWithoutTokenAddress.join(","),
+      tokenList
+    );
```

Comment:
nit: I would try to avoid string concatenation at much as possible, we could use a join for the new lines:

```js
[
    headerRow,
    rowWithoutDecimal.join(","),
    rowWithDecimal.join(","),
    rowWithoutTokenAddress.join(",")
].join("\n")
```

but if this is going to be used more than once in the tests, maybe we could make a method `makeCSV` or something for this where we just pass in the rows.